Libs/MRML/Logic/vtkMRMLSliceLogic.cxx

```diff
@@ -1022,6 +1033,706 @@ void vtkMRMLSliceLogic::UpdateReconstructionSlab(vtkMRMLSliceLogic* sliceLogic,
+    vtkMath::Normalize(transformGridAxisX);
+  }
+
+  // Rotate by rotationDeg around the Z axis
+  vtkSmartPointer<vtkMatrix4x4> gridDirectionMatrix = vtkSmartPointer<vtkMatrix4x4>::New();
+  gridDirectionMatrix->Identity();
+  for (int i = 0; i < 3; ++i)
+  {
+    gridDirectionMatrix->SetElement(i, 0, transformGridAxisX[i]);
+    gridDirectionMatrix->SetElement(i, 1, transformGridAxisY[i]);
+    gridDirectionMatrix->SetElement(i, 2, transformGridAxisZ[i]);
+  }
+  //
+  vtkSmartPointer<vtkTransform> gridDirectionTransform = vtkSmartPointer<vtkTransform>::New();
+  gridDirectionTransform->Concatenate(gridDirectionMatrix);
+  gridDirectionTransform->RotateZ(rotationDeg);
+  //
+  vtkSmartPointer<vtkMatrix4x4> rotatedGridMatrix = vtkSmartPointer<vtkMatrix4x4>::New();
+  gridDirectionTransform->GetMatrix(rotatedGridMatrix);
+  for (int i = 0; i < 3; ++i)
+  {
+    transformGridAxisX[i] = rotatedGridMatrix->GetElement(i, 0);
+    transformGridAxisY[i] = rotatedGridMatrix->GetElement(i, 1);
+    transformGridAxisZ[i] = rotatedGridMatrix->GetElement(i, 2);
+  }
+
+  if (stretching)
+  {
+    // Project curve points to grid YZ plane
+    vtkSmartPointer<vtkMatrix4x4> transformFromGridYZPlane = vtkSmartPointer<vtkMatrix4x4>::New();
+    transformFromGridYZPlane->Identity();
+    const double* origin = curveNodePlane->GetOrigin();
+    for (int i = 0; i < 3; ++i)
+    {
+      transformFromGridYZPlane->SetElement(i, 0, transformGridAxisY[i]);
+      transformFromGridYZPlane->SetElement(i, 1, transformGridAxisZ[i]);
+      transformFromGridYZPlane->SetElement(i, 2, transformGridAxisX[i]);
+      transformFromGridYZPlane->SetElement(i, 3, origin[i]);
+    }
+    vtkSmartPointer<vtkMatrix4x4> transformToGridYZPlane = vtkSmartPointer<vtkMatrix4x4>::New();
+    vtkMatrix4x4::Invert(transformFromGridYZPlane, transformToGridYZPlane);
+
+    vtkPoints* originalCurvePointsArray = curveNode->GetCurvePoints();
+    vtkSmartPointer<vtkPoints> curvePointsProjected_RAS = vtkSmartPointer<vtkPoints>::New();
+    this->CurvedPlanarReformationGetPointsProjectedToPlane(
+      originalCurvePointsArray, transformToGridYZPlane, curvePointsProjected_RAS);
+    for (int i = resampledCurveNode->GetNumberOfControlPoints() - 1; i >= 0; --i)
+    {
+      resampledCurveNode->RemoveNthControlPoint(i);
+    }
+    for (int i = 0; i < curvePointsProjected_RAS->GetNumberOfPoints(); ++i)
+    {
+      resampledCurveNode->AddControlPoint(curvePointsProjected_RAS->GetPoint(i));
+    }
+
+    // After projection, resampling is needed to get uniform distances
+    originalCurvePoints = resampledCurveNode->GetCurvePointsWorld();
+    sampledPoints = vtkSmartPointer<vtkPoints>::New();
+    if (!vtkMRMLMarkupsCurveNode::ResamplePoints(originalCurvePoints, sampledPoints, resamplingCurveSpacing, false))
+    {
+      vtkErrorMacro("vtkMRMLSliceLogic::CurvedPlanarReformationComputeStraighteningTransform failed: "
+                    "second call to resampling curve failed");
```

Comment:
Since the `vtkErrorMacro` already display the fact there is an error, the file name & line. All messages should be simplified like I already suggested earlier.

Reply:
I don't see your `vtkErrorMacro` suggestion, but I have nonetheless simplified the messages for the CurvedPlanarReformation methods.

Reply:
You need to click on the the following to see all suggestions:

![image](https://github.com/user-attachments/assets/734044b4-61ae-414d-8427-de1cc278f8e6)

Reply:
Thank you